File: vetix/plugins/rare_file.py
from vetix.plugin import Issue, Plugin, Severity
from vetix.utils.file_classifier import is_risk_file
from vetix.utils.utils import get_relative_path


class RareFileCheckPlugin(Plugin):
    """SKILL documents that may contain rare auxiliary files."""

    name = "rare file"

    def scan(self, skill_dir: str, file_path: str, content: str) -> list[Issue]:
        if not is_risk_file(file_path):
            return []

        relative_path = get_relative_path(file_path, skill_dir)
        return [Issue(
            name="Rare file",
            severity=Severity.MEDIUM,
            description="The SKILL directory contains rare auxiliary files that may pose a security risk.",
            file_path=relative_path,
            audit_required=False
        )]

File: internal/plugin/rare_file.go
package plugin

import "vetix/internal/pluginutils"

type RareFileCheckPlugin struct{}

func (RareFileCheckPlugin) Meta() Meta {
	return Meta{
		ID:          "rare_file",
		Name:        "Rare File Extension",
		Description: "Flags files whose extension is outside the known text-file whitelist.",
	}
}

func (RareFileCheckPlugin) Scan(skillDir, filePath, content string) []Issue {
	if !pluginutils.IsRiskFile(filePath) {
		return nil
	}
	return []Issue{{
		Name:          "Rare file",
		Severity:      SeverityMedium,
		Category:      CatObfuscation,
		Description:   "The SKILL directory contains rare auxiliary files that may pose a security risk.",
		FilePath:      relativePath(filePath, skillDir),
		AuditRequired: false,
	}}
}
